src/buffer/buffer.rs

```diff
@@ -171,38 +204,102 @@ impl Buffer {
     /// buffer.pos_of(100); // Panics
     /// ```
     pub fn pos_of(&self, i: usize) -> (u16, u16) {
-        debug_assert!(
-            i < self.content.len(),
-            "Trying to get the coords of a cell outside the buffer: i={i} len={}",
-            self.content.len()
-        );
+        self.try_pos_of(i).unwrap_or_else(|| {
+            panic!(
+                "Trying to get the coords of a cell outside the buffer: i={i} len={}",
+                self.content.len()
+            )
+        })
+    }
+
+    /// Returns an option containing the (global) coordinates of a cell given its index
+    /// or `None` when given an index that is outside the Buffer's content.
+    /// Global coordinates are offset by the Buffer's area offset (`x`/`y`).
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use ratatui::prelude::*;
+    /// let rect = Rect::new(200, 100, 10, 10);
+    /// let buffer = Buffer::empty(rect);
+    /// assert_eq!(buffer.pos_of(0), (200, 100));
+    /// assert_eq!(buffer.pos_of(14), (204, 101));
+    /// ```
+    #[must_use]
+    pub fn try_pos_of(&self, i: usize) -> Option<(u16, u16)> {
+        (i < self.content.len()).then_some(Self::calculate_coordinates_from_index(self.area, i))
+    }
+    const fn calculate_coordinates_from_index(area: Rect, i: usize) -> (u16, u16) {
         (
-            self.area.x + (i as u16) % self.area.width,
-            self.area.y + (i as u16) / self.area.width,
+            area.x + (i as u16) % area.width,
+            area.y + (i as u16) / area.width,
         )
     }
 
     /// Print a string, starting at the position (x, y)
+    /// # Panics
+    ///
+    /// Panics when given an coordinate that is outside of this Buffer's area.
     pub fn set_string<T, S>(&mut self, x: u16, y: u16, string: T, style: S)
     where
         T: AsRef<str>,
         S: Into<Style>,
     {
-        self.set_stringn(x, y, string, usize::MAX, style);
+        self.try_set_stringn(x, y, string, usize::MAX, style.into())
+            .unwrap_or_else(|| Self::panic_on_wrong_position(x, y, self.area));
+    }
+
+    /// Print a string, starting at the position (x, y)
+    ///
+    /// `style` accepts any type that is convertible to [`Style`] (e.g. [`Style`], [`Color`], or
+    /// your own type that implements [`Into<Style>`]).
+    #[must_use]
+    pub fn try_set_string<T, S>(&mut self, x: u16, y: u16, string: T, style: S) -> Option<()>
```

Comment:
I'd like to eventually deprecate these methods and use Line/Span render instead.